tenacity/__init__.py

```diff
@@ -95,153 +83,32 @@ def wrapped_f(*args, **kw):
         return wrap
 
 
-class TryAgain(Exception):
-    """Always retry the executed function when raised."""
-
-
 class Retrying(object):
     """Retrying controller."""
 
-    def __init__(self,
-                 stop=stop_never, wait=wait_none(),
-                 sleep=time.sleep,
-                 retry=retry_if_exception_type(),
-                 before=before_nothing,
-                 after=after_nothing,
-                 reraise=False):
+    def __init__(self, marshaller, sleep=time.sleep):
+        self.marshaller = marshaller
         self.sleep = sleep
-        self.stop = stop
-        self.wait = wait
-        self.retry = retry
-        self.before = before
-        self.after = after
-        self.reraise = reraise
-        self._local = threading.local()
-
-    def __repr__(self):
-        attrs = _utils.visible_attrs(self, attrs={'me': id(self)})
-        return ("<Retrying object at 0x%(me)x (stop=%(stop)s, "
-                "wait=%(wait)s, sleep=%(sleep)s, retry=%(retry)s, "
-                "before=%(before)s, after=%(after)s)>") % (attrs)
-
-    @property
-    def statistics(self):
-        """A dictionary of runtime statistics this controller has gathered.
-
-        This dictionary will be empty when the controller has never been
-        ran. When it is running or has ran previously it should have (but
-        may not) have useful and/or informational keys and values when
-        running is underway and/or completed.
-
-        .. warning:: The keys in this dictionary **should** be some what
-                     stable (not changing), but there existence **may**
-                     change between major releases as new statistics are
-                     gathered or removed so before accessing keys ensure that
-                     they actually exist and handle when they do not.
-
-        .. note:: The values in this dictionary are local to the thread
-                  running call (so if multiple threads share the same retrying
-                  object - either directly or indirectly) they will each have
-                  there own view of statistics they have collected (in the
-                  future we may provide a way to aggregate the various
-                  statistics from each thread).
-        """
-        try:
-            return self._local.statistics
-        except AttributeError:
-            self._local.statistics = {}
-            return self._local.statistics
+        self.statistics = marshaller.statistics
```

Comment:
Hm rather than copy the ref (I guess it works because it's a dict) maybe go for a `@property`.